egs/wsj/s5/steps/libs/nnet3/xconfig/basic_layers.py

```diff
@@ -343,7 +343,7 @@ def get_full_config(self):
         # unlike other layers the input layers need to be printed in
         # 'init.config' (which initializes the neural network prior to the LDA)
         ans = []
-        for config_name in [ 'init', 'ref', 'final' ]:
+        for config_name in [ 'ref', 'final' ]:
```

Comment:
Hi, @danpovey , thank you for your detailed advice! But i think this will have problems when one situation that there is one or more FixedAffineLayer, which is not for LDA, just for feat transform for example. In this situation, we shouldn't do LDA also. I think the key is the FixedAffineLayer is not for LDA usage only, so how to decide the FixedAffineLayer is for LDA or not, i come up with the method using the 'name=lda' indication.

Maybe something that i misunderstanded, this is the first time i commit code to kaldi, and i hope it will be more in future. I'm very glad to communicate with you @danpovey ! I'm in china, and it's late in Baltimore. Thank you always replay as soon as you can!